server/plugin.ts

```diff
@@ -133,7 +145,7 @@ export class SecurityPlugin implements Plugin<SecurityPluginSetup, SecurityPlugi
     }
 
     // Register server side APIs
-    defineRoutes(router);
+    defineRoutes(router, dataSourceEnabled);
```

Comment:
do you think we can create and utilize a setting `this.config.datasource.enabled` throughout instead of passing `dataSourceEnabled` ?

Reply:
This function is registering some routes on the router, and doesn't have the context of the plugin inside the function. We need to pass it in somehow, I think this is fine for now.